cmd/dagger/flags.go

```diff
@@ -354,7 +354,7 @@ func (v *directoryValue) Get(ctx context.Context, dag *dagger.Client, modSrc *da
 }
 
 // makeGitDirectory creates a dagger.Directory object from a parsed gitutil.GitURL
-func makeGitDirectory(gitURL *gitutil.GitURL, dag *dagger.Client) *dagger.Directory {
+func makeGitDirectory(gitURL *gitutil.GitURL, dag *dagger.Client, ignorePattern []string) *dagger.Directory {
```

Comment:
This adds an extra responsibility to the function, I think the function should just turn a URL into a Directory, so I suggest filtering in the flag instead.